backend/pkg/api/connect/service/monitoring/v1/service.go

```diff
@@ -0,0 +1,207 @@
+	val, ok := map[int32]v1.KafkaAPI{
+		0:  v1.KafkaAPI_KAFKA_API_PRODUCE,
+		1:  v1.KafkaAPI_KAFKA_API_FETCH,
+		2:  v1.KafkaAPI_KAFKA_API_OFFSETS,
+		3:  v1.KafkaAPI_KAFKA_API_METADATA,
+		4:  v1.KafkaAPI_KAFKA_API_LEADER_AND_ISR,
+		5:  v1.KafkaAPI_KAFKA_API_STOP_REPLICA,
+		6:  v1.KafkaAPI_KAFKA_API_UPDATE_METADATA,
+		7:  v1.KafkaAPI_KAFKA_API_CONTROLLED_SHUTDOWN,
+		8:  v1.KafkaAPI_KAFKA_API_OFFSET_COMMIT,
+		9:  v1.KafkaAPI_KAFKA_API_OFFSET_FETCH,
+		10: v1.KafkaAPI_KAFKA_API_GROUP_COORDINATOR,
+		12: v1.KafkaAPI_KAFKA_API_JOIN_GROUP,
+		13: v1.KafkaAPI_KAFKA_API_HEARTBEAT,
+		14: v1.KafkaAPI_KAFKA_API_LEAVE_GROUP,
+		15: v1.KafkaAPI_KAFKA_API_SYNC_GROUP,
+		16: v1.KafkaAPI_KAFKA_API_DESCRIBE_GROUPS,
+		17: v1.KafkaAPI_KAFKA_API_LIST_GROUPS,
+		18: v1.KafkaAPI_KAFKA_API_SASL_HANDSHAKE,
+		19: v1.KafkaAPI_KAFKA_API_API_VERSIONS,
+		20: v1.KafkaAPI_KAFKA_API_CREATE_TOPICS,
+		21: v1.KafkaAPI_KAFKA_API_DELETE_TOPICS,
+	}[key]
+
+	if !ok {
+		return v1.KafkaAPI_KAFKA_API_UNSPECIFIED
+	}
+	return val
+}
+
+func adminConnectionToConnect(conn *adminv2.KafkaConnection) *v1.Connection {
+	// Convert the current requests into our format
+	currentRequests := make([]*v1.ActiveRequests_Request, len(conn.InFlightRequests.SampledInFlightRequests))
+	for i, req := range conn.InFlightRequests.SampledInFlightRequests {
+		currentRequests[i] = &v1.ActiveRequests_Request{
+			Api:      kafkaAPIFromKey(req.ApiKey),
+			Duration: req.InFlightDuration,
+		}
+	}
+
+	apiVersions := []*v1.APIVersion{}
+	for key, ver := range conn.ApiVersions {
+		apiVersions = append(apiVersions, &v1.APIVersion{
+			Api:     kafkaAPIFromKey(key),
+			Version: ver,
+		})
+	}
+
+	// Calculate how long the connection has been open
+	opened := conn.OpenTime.AsTime()
+	closed := conn.CloseTime.AsTime()
+	duration := closed.Sub(opened)
+	if opened.After(closed) {
+		duration = time.Since(opened)
+	}
+
+	return &v1.Connection{
+		NodeId:             conn.NodeId,
+		ShardId:            conn.ShardId,
+		Uid:                conn.Uid,
+		State:              conn.State,
+		Authentication:     conn.AuthenticationInfo,
+		OpenTime:           conn.OpenTime,
+		CloseTime:          conn.CloseTime,
+		ConnectionDuration: durationpb.New(duration),
+		IdleDuration:       conn.IdleDuration,
+		TlsEnabled:         conn.TlsInfo.Enabled,
+		Client: &v1.ConnectionClient{
+			Ip:              conn.Source.IpAddress,
+			Port:            conn.Source.Port,
+			Id:              conn.ClientId,
+			SoftwareName:    conn.ClientSoftwareName,
+			SoftwareVersion: conn.ClientSoftwareVersion,
+		},
+		Group: &v1.GroupInfo{
+			Id:         conn.GroupId,
+			InstanceId: conn.GroupInstanceId,
+			MemberId:   conn.GroupMemberId,
+		},
+		ListenerName:    conn.ListenerName,
+		TransactionalId: conn.TransactionalId,
+		ApiVersions:     apiVersions,
+		ActiveRequests: &v1.ActiveRequests{
+			Requests:        currentRequests,
+			HasMoreRequests: conn.InFlightRequests.HasMoreRequests,
```

Comment:
InflightRequests also worth double confirming if this is never nil or we will immediately have panics. Couldn't inspect the adminv2 types from the github PR view, sorry :D